src/useVesselLink.ts

```diff
@@ -31,14 +28,23 @@ export default function useVesselLink(config: ClientConfig) {
     linkToken: string;
     integrationId?: string;
   }) => {
-    if (popupLoaded && !error && window.Vessel) {
-      window.Vessel.open({ integrationId, linkToken });
+    if (error) {
+      throw new Error(`Error loading Vessel script: ${error}`);
+    }
+    if (!window.Vessel) {
+      console.error('Vessel is not initialized');
+      return;
+    }
+    if (!window.Vessel.state?.isLoaded) {
+      console.error('Vessel has not been loaded, did you call Vessel.init()?');
+      return;
     }
+    // Open modal
+    window.Vessel.open({ integrationId, linkToken });
   };
 
   return {
     error,
     open,
-    loading: !popupLoaded,
```

Comment:
Do we not need to return this anymore? Guessing this is a prop customers might use?

Reply:
Yeah this was picked up form an earlier PR but hasn't been released yet so it won't have been used by anyone. 

Give me a second to see if I can come up with a way for this to be surfaced

Reply:
Yeah rip, my React knowledge is hitting it's limit here. Going to leave for now and investigate in the near future.